### PR TITLE
Start hlsl-out

### PR DESCRIPTION
**Pull request by Napokue**
Start implementation of hlsl-out. This will cover only the empty case. When this is landed a simple empty entrypoint can be generated via the hlsl-out path.

### CODE REVIEW

File: src/back/hlsl/mod.rs [hunk 1]
```diff
@@ -0,0 +1,22 @@
+mod writer;
+
+use std::string::FromUtf8Error;
+pub use writer::Writer;
+
+#[derive(Debug)]
+pub enum Error {
+    Utf8(FromUtf8Error),
```
Comment (kvark) [inline src/back/hlsl/mod.rs:8]: you may use `thiserror` from the start as the better practice

File: src/back/hlsl/writer.rs [hunk 2]
```diff
@@ -0,0 +1,20 @@
+use crate::back::hlsl::Error;
```
Comment (kvark) [inline src/back/hlsl/writer.rs:1]: I don't think any backend should include anything from either `crate::back` or `crate::front`. Instead, let's do `use super::Error`.

Comment (Napokue) [reply src/back/hlsl/writer.rs:1]: Yes agreed, probably an auto complete function in my IDE that auto imports.

**Comment by Napokue** [comment]
The scope of just naming this PR: "Implementation hlsl-out" was a bit too ambitious for me to undertake. I have down scaled the scope of this PR, and this PR will add the bare minimum to at least enable the hlsl-out path.

**Comment by Napokue** [comment]
The PR should be ready, the only thing I don't know how to do yet is plugging this into our test framework with the snapshots.

**Comment by kvark** [comment]
Please make CI happy :)
I assume it's fine to squash it on merge, right?
> only thing I don't know how to do yet is plugging this into our test framework with the snapshots.

You can just follow MSL and GLSL in the `snapshots.rs`.

**Comment by Napokue** [comment]
> I assume it's fine to squash it on merge, right?
Sure.

I think I have added hlsl-out to the snapshots. I only can't seem to find how the output snapshots are generated, or are they created by hand?

**Comment by kvark** [comment]
> I only can't seem to find how the output snapshots are generated, or are they created by hand?

they are created by `cargo test` or `make`.

**Comment by Napokue** [comment]
Found it. This `cargo insta` is a pretty handy tool. I created them with `cargo insta review`

**Review by kvark — APPROVED** [review]

**Pull request merged**